doc/dev/doc/events.md

```diff
@@ -39,25 +41,110 @@ private async ValueTask OnThingUpdated(Thing sender, ThingUpdatedEventArgs args)
+
+### PartitionsUpdated
+
+The `PartitionsUpdated` event triggers whenever the server notifies the client of a new partitions list. This happens when the partitions list changes, but also periodically when the server wants to ensure that clients are aware of partitions. Handlers do not receive any event arguments.
+
+### MembersUpdated
+
+The `MembersUpdated` event triggers whenever the server notifies the client of a new members list. This happens when members are added or removed from the cluster, but also periodically when the server wants to ensure that clients know about members. Handlers receive an instance of the @Hazelcast.Events.MembersUpdatedEventArgs class, which exposes the following properties:
+* @Hazelcast.Events.MembersUpdatedEventArgs.AddedMembers: a collection of @Hazelcast.Models.MemberInfo representing the members that were added to the cluster
+* @Hazelcast.Events.MembersUpdatedEventArgs.RemovedMembers: a collection of @Hazelcast.Models.MemberInfo representing the members that were removed from the cluster
+* @Hazelcast.Events.MembersUpdatedEventArgs.Members: a collection of @Hazelcast.Models.MemberInfo representing all members in the cluster
+
+### ObjectCreated
+
+The `ObjectCreated` event triggers whenever the server notifies the client that a new distributed object has been created (for instance, when the server creates a new map named `my-map`). Handlers receive an instance of the @Hazelcast.Events.DistributedObjectCreatedEventArgs class, which exposes the following properties:
+* @Hazelcast.Events.DistributedObjectLifecycleEventArgs.ServiceName: the internal Hazelcast service name (for instance, for maps, `hz:impl:mapService`)
+* @Hazelcast.Events.DistributedObjectLifecycleEventArgs.Name: the name of the created object (for instance, `my-map`)
+* @Hazelcast.Events.DistributedObjectLifecycleEventArgs.SourceMemberId: the identifier of the member which triggered the event
+
+### ObjectDestroyed
+
+The `ObjectDestroyed` event triggers whenever the server notifies the client that a distributed object has been destroyed (for instance, when the client requests that the server destroys a map named `my-map`). Handlers receive an instance of the @Hazelcast.Events.DistributedObjectDestroyedEventArgs class, which exposes the following properties:
+* @Hazelcast.Events.DistributedObjectLifecycleEventArgs.ServiceName: the internal Hazelcast service name (for instance, for maps, `hz:impl:mapService`)
+* @Hazelcast.Events.DistributedObjectLifecycleEventArgs.Name: the name of the destroyed object (for instance, `my-map`)
+* @Hazelcast.Events.DistributedObjectLifecycleEventArgs.SourceMemberId: the identifier of the member which triggered the event
 
 ## Distributed Objects Events
 
-TODO: complete
\ No newline at end of file
+Each type of distributed object exposes events specific to the type. For instance, @Hazelcast.DistributedObjects.IHList`1 exposes the `ItemAdded` event:
```

Comment:
Backtick in generic type needs to be escaped or it messes with the code.
```suggestion
Each type of distributed object exposes events specific to the type. For instance, @Hazelcast.DistributedObjects.IHList\`1 exposes the `ItemAdded` event:
```

Reply:
ok

Reply:
Will @ symbol work as intended with backtick in type name?

Reply:
OK, IHList\`1 does not work, I have removed the backslash
And then, why did we need this backslash in the first place?

Reply:
Because of markdown, check _events.md_ preview:
![image](https://user-images.githubusercontent.com/4650832/131814324-466bd05e-f9f2-4b5c-b717-221b7253d8ce.png)

Reply:
ah - but that's only because the markdown editor is confused. when you build the docs it works.